### PR TITLE
Add P2 autocomplete to block editor

### PR DESCRIPTION
**Pull request by marekhrabe**
#### Changes proposed in this Pull Request

~Needs D32779-code.~ _(Deployed)_

See p6OVho-fw-p2 for the original project proposal, including mockups illustrating the feature.

#### Testing instructions

_Apply D32832-code to your sandbox, and sandbox a P2. See that diff's description for more detailed instructions._

D32832-code generated by doing the following:

- Find the `ci/circleci: build-o2-blocks` check at the bottom of this PR and click on "Details" to find the job number.
- Log in to your sandbox and fetch the updated block code with the bin script specified in #gm-19-project-p2-auto-suggestions

### CODE REVIEW

**Comment by kwight** [comment]
@marekhrabe I'll put together an endpoint we can use for this 👍

**Comment by kwight** [comment]
Oh sorry! Just noticed the "in progress" label – anyhoo, I'll hack together a quick endpoint for you to use.

**Comment by marekhrabe** [comment]
We have an endpoint ready and we'll be connecting it. Don't worry about that huge list. It's for our dev purposes and was never intended to be merged. It was once nice and compact line on my dev machine but the code formatter has made it into this when committing

**Comment by kwight** [comment]
@marekhrabe lol sorry – got an email as "owner" for review and dived in – continue on! 🙃

File: apps/o2-blocks/src/p2-autocomplete/editor.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import { addFilter } from '@wordpress/hooks';
+import { map, unescape } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import './editor.scss';
+
+/*
+ * This is a workaround for the way Gutenberg autocomplete works. It onlylooks for
```
Comment (ockham) [inline apps/o2-blocks/src/p2-autocomplete/editor.js]:
```suggestion
 * This is a workaround for the way Gutenberg autocomplete works. It only looks for
```

File: apps/o2-blocks/src/p2-autocomplete/editor.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		<div className="p2-autocomplete">
+			<span key="subdomain" className="p2-autocomplete__subdomain">
+				+{ site.subdomain }
+			</span>
+			<span key="description" className="p2-autocomplete__title">
+				{ unescape( site.title ) }
+			</span>
+			{ site.blavatar ? (
+				<img
+					key="blavatar"
```
Comment (ockham) [inline apps/o2-blocks/src/p2-autocomplete/editor.js]:
```suggestion
```

File: apps/o2-blocks/src/p2-autocomplete/editor.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+/*
+ * This is a workaround for the way Gutenberg autocomplete works. It only looks for
+ * beginnings of words and something like "teamabcp2" won't be found if you search "abc".
+ * Adding spaces around the word solves the problem.
+ */
+const COMMON_PREFIXES = /(team|a8c|woo|happiness)/i;
+const stripCommonWords = str => {
+	return str.replace( COMMON_PREFIXES, ' ' );
+};
```
Comment (ockham) [inline apps/o2-blocks/src/p2-autocomplete/editor.js]: Could do something like
```
const stripCommonWords = str => str.replace( COMMON_PREFIXES, ' ' );
```

Not that it mattered much.

**Pull request merged**